app/models/custom_button.rb

```diff
@@ -15,6 +15,8 @@ class CustomButton < ApplicationRecord
   validates :name, :description, :uniqueness => {:scope => [:applies_to_class, :applies_to_id]}, :presence => true
   validates :guid, :uniqueness => true, :presence => true
 
+  after_destroy :cleanup_resource_actions
```

Comment:
The relation is already `:dependent => :destroy` https://github.com/ManageIQ/manageiq/pull/18152/files#diff-72d768ae2e73b9e77080d4a7df59cf56R2

Reply:
Okay but the error information is still pretty useful.